File: src/0791/S0791_v1.py
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        cnt = [0] * 26
        for i in s:
            cnt[ord(i)-97] += 1
        res = ''
        for j in order:
            while cnt[ord(j)-97] != 0:
                res += j
                cnt[ord(j)-97] -= 1
        for i, v in enumerate(cnt):
            while v != 0:
                res += chr(i+97)
                v -= 1
        return res

File: src/0791/S0791_v1.c
char * customSortString(char * order, char * s){
    int cnt[123];
    for (int i=97;i<123;++i) cnt[i] = 0;
    for (int i=0;i<strlen(s);++i) cnt[s[i]]++;
    int p = 0;
    for (int i=0;i<strlen(order);++i) {
        while (cnt[order[i]]) {
            s[p++] = order[i];
            cnt[order[i]]--;
        }
    }
    for (int i=97;i<123;++i) while (cnt[i]--) s[p++] = i;
    return s;
}
